app/models/item_change_set.rb

```diff
@@ -2,56 +2,36 @@
 
 # Represents a set of changes to an item.
 class ItemChangeSet
+  PROPERTIES = %i[
+    admin_policy_id
+    catkey
+    collection_ids
+    copyright_statement
+    license
+    source_id
+    use_statement
+  ].freeze
+
   def initialize
     @changes = {}
     yield self
   end
 
-  def catkey=(key)
-    @changes[:catkey] = key
-  end
-
-  def catkey
-    @changes[:catkey]
-  end
-
-  def catkey_changed?
-    @changes.key?(:catkey)
-  end
-
-  def collection_ids=(ids)
-    @changes[:collection_ids] = ids
-  end
-
-  def collection_ids
-    @changes[:collection_ids]
-  end
+  PROPERTIES.each do |property|
+    define_method(property) do
+      @changes[property]
+    end
 
-  def collection_ids_changed?
-    @changes.key?(:collection_ids)
-  end
-
-  def source_id=(id)
-    @changes[:source_id] = id
-  end
-
-  def source_id
-    @changes[:source_id]
-  end
-
-  def source_id_changed?
-    @changes.key?(:source_id)
-  end
-
-  def admin_policy_id=(id)
-    @changes[:admin_policy_id] = id
-  end
+    define_method("#{property}_changed?") do
+      @changes.key?(property)
+    end
 
-  def admin_policy_id
-    @changes[:admin_policy_id]
+    define_method("#{property}=") do |value|
+      @changes[property] = value
+    end
   end
 
-  def admin_policy_id_changed?
-    @changes.key?(:admin_policy_id)
+  def changed?
```

Comment:
can you drop a comment to say this is for the whole object?  Or is that silly?  (ditto for collection)

Reply:
will do!